manifests/file.pp

```diff
@@ -35,7 +35,7 @@
     $target_file = $target
     $staging_dir = staging_parse($target, 'parent')
   } else {
-    $staging_dir = "${staging::path}/${subdir}"
+    $staging_dir = regsubst("${staging::_path}/${subdir}", '/$', '') # Strip off trailing slashes
```

Comment:
There are two possible cases here.
* `$subdir` is an empty string (happens if there is no calling_module)
* `$subdir` was provided with a trailing slash (much less likely).